src/test-run/index.ts

```diff
@@ -775,6 +783,159 @@ export default class TestRun extends AsyncEventEmitter {
         return consoleMessageCopy[String(this.activeWindowId)];
     }
 
+    private async _findCookiesByApi (urls: { domain: string; path: string }[], key?: string): Promise<(InternalCookie | InternalCookie[])[]> {
```

Comment:
I propose to pack the cookie-related API to the `CookieManagment` class that will be initialized with the `this.session.cookies`. It allows to cover all functionality in the server tests and reduce the functional tests to 3 items: `SetCookie`, `GetCookie`, and `deleteCookie`.

Reply:
I moved this functionality to the hammerhead and add tests for one there.